front/components/agent_builder/triggers/TriggerSelectionPage.tsx

```diff
@@ -0,0 +1,141 @@
+  onClose: () => void;
+  onScheduleSelect: () => void;
+  onWebhookSelect: (webhookSourceView: WebhookSourceViewType) => void;
+  webhookSourceViews: WebhookSourceViewType[];
+}
+
+export function TriggerSelectionPage({
+  isOpen,
+  onClose,
+  onScheduleSelect,
+  onWebhookSelect,
+  webhookSourceViews,
+}: TriggerSelectionPageProps) {
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const filteredWebhookSourceViews = useMemo(() => {
+    if (!searchTerm.trim()) {
+      return webhookSourceViews;
+    }
+
+    const term = searchTerm.toLowerCase();
+    return webhookSourceViews.filter((view) => {
+      return [view.customName, view.description].some((field) =>
+        field?.toLowerCase().includes(term)
+      );
+    });
+  }, [searchTerm, webhookSourceViews]);
+
+  // Adding a few useful shortcut keywords for schedules.
+  const showSchedule = useMemo(() => {
+    if (!searchTerm.trim()) {
+      return true;
+    }
+
+    const term = searchTerm.toLowerCase();
+    return (
+      "schedule".includes(term) ||
+      "time".includes(term) ||
+      "cron".includes(term)
+    );
+  }, [searchTerm]);
+
+  return (
+    <Sheet open={isOpen} onOpenChange={(open) => !open && onClose()}>
+      <SheetContent size="lg">
+        <SheetHeader>
+          <SheetTitle>Add triggers</SheetTitle>
+        </SheetHeader>
+        <SheetContainer>
+          <SearchInput
+            placeholder="Search triggers..."
+            value={searchTerm}
+            onChange={setSearchTerm}
+            name="triggerSearch"
+            className="mt-4"
+          />
+
+          <div className="flex flex-col gap-4 py-2">
+            {showSchedule && (
+              <>
+                <span className="text-lg font-semibold">Top triggers</span>
+                <div className="grid grid-cols-2 gap-3">
+                  <ToolCard
+                    icon={TimeIcon}
+                    label="Schedule"
+                    description="Trigger this agent on a schedule"
+                    isSelected={false}
+                    canAdd={true}
+                    onClick={onScheduleSelect}
+                    cardContainerClassName="h-36"
+                  />
+                </div>
+              </>
+            )}
+
+            {filteredWebhookSourceViews.length > 0 && (
+              <div className="flex flex-col gap-4">
+                <span className="text-lg font-semibold">Other triggers</span>
+                <div className="grid grid-cols-2 gap-3">
+                  {filteredWebhookSourceViews.map((view) => {
+                    const icon = isCustomResourceIconType(view.icon)
+                      ? ActionIcons[view.icon]
+                      : InternalActionIcons[view.icon];
+
+                    return (
+                      <ToolCard
+                        key={view.sId}
+                        icon={icon}
+                        label={view.customName}
+                        description={view.description}
```

Comment:
we either need to force entering a description in the trigger section of the space administration as you said, or find a good default here (i.e `Trigger your agent with ${view.customname}.` 


```suggestion
                        description={view.description ?? `Trigger this agent with ${view.customname}.`}
```